frontend/readline.c

```diff
@@ -37,10 +43,39 @@ static void sigint_handler(int n) {
 }
 #endif
 
+
+static void mkdirs(const char *path) {
+	struct stat st = {0};
+	if (stat(path, &st) == 0 && S_ISDIR(st.st_mode))
+		return;
+
+	char *subpath;
+
+	subpath = strdup(path);
+	dirname(subpath);
```

Comment:
`dirname` returns a pointer to the actual dirname. It may store it in `subpath`, but may not. We shouldn't rely on `dirname` modifying `subpath`.